test/OptionSettlementEngine.invariant.t.sol

```diff
@@ -17,12 +17,15 @@ contract OptionSettlementEngineInvariantTest is BaseEngineTest, InvariantTest {
     ProtocolAdmin internal admin;
     Timekeeper internal timekeeper;
 
+    uint256[] internal optionTypes;
```

Comment:
Really nice. I see how we'll be able to track what we need to, and how we can calculate invariants using totals and other variables which may not be tracked by Engine.

We might want to refactor out these arrays into a World object, instead of the test contract itself. This will give us flexibility to use in other test harnesses and it will remove a circular dependency — we would pass the _World_ to each Actor instead of the test suite and include it as a state variable in the test contract (kinda like Timekeeper).

LMK what you think, I'm sure there are trade-offs. Separating out the tracking / accumulating into a world object is a pattern from BDD style testing that I think fits what you're doing well and may open doors for us for some other good techniques too.

Reply:
agree with the idea, was mostly just working around for the sake of this PR, and I think that would be a good refactor. 

but how much more test code are we going to be authoring which would be able to leverage your proposed structure? if the answer is "a lot" then we should definitely hit the refac. but, are we actually going to be writing new test harnesses at this point? should we really invest more hours into refactoring test code?

I'm skeptical that whatever additional tests we're going to write (or not write) would justify spending more time on refactors. The erc20 wrappers seem like a much higher priority.

Reply:
Sounds good, let's hold on World for time being. Maybe will be helpful for Wrapper, Margin, Vaults, etc. Let's wait and see.